### PR TITLE
olatechpro:feature/output-variable-escaping

### PR DESCRIPTION
**Pull request by olatechpro**
## Description
Variable escaping of checklist output

## Benefits
Previously unescaped checklist output are now escaped

## Checklist

- [x] I have created a specific branch for this pull request before committing, starting off the current HEAD of `development` branch. 
- [x] I'm submitting to the `development`, feature/hotfix/release branch. (Do not submit to the master branch!)
- [ ] This pull request relates to a specific problem (bug or improvement).
- [ ] I have mentioned the issue number in the pull request description text.
- [ ] All the issues mentioned in this pull request relate to the problem I'm trying to solve.
- [x] The code I'm sending follows the [PSR-12](https://www.php-fig.org/psr/psr-12/) coding style.

### CODE REVIEW

File: modules/settings/templates/settings.php [hunk 1]
```diff
@@ -6,22 +6,22 @@
                     <?php foreach ($context['modules'] as $module) : ?>
                         <?php if (!empty($context['module']['options_page']) && $context['module']->options->enabled === 'on') : ?>
                             <a
-                                    href="?page=<?php echo $context['slug']; ?>&module=<?php echo $context['module']->settings_slug; ?>"
+                                    href="?page=<?php echo esc_attr($context['slug']); ?>&module=<?php echo esc_attr($context['module']->settings_slug); ?>"
```
Comment (andergmartins) [inline modules/settings/templates/settings.php:9]: Please, add all the URL in a `esc_url` function, removing the `esc_attr` function.

**Review by andergmartins — CHANGES_REQUESTED** [review]

File: modules/checklists/templates/footer.php [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
-        $plugin_name_markup  = '<strong>' . $context['plugin_name'] . '</strong>';
+        $plugin_name_markup  = '<strong>' . esc_html($context['plugin_name']) . '</strong>';
         $rating_stars_markup = '<span class="dashicons dashicons-star-filled"></span><span class="dashicons dashicons-star-filled"></span><span
                 class="dashicons dashicons-star-filled"></span><span
                 class="dashicons dashicons-star-filled"></span><span class="dashicons dashicons-star-filled"></span>';
         ?>
 
         <a
-                href="//wordpress.org/support/plugin/<?php echo $context['plugin_slug']; ?>/reviews/#new-post"
+                href="//wordpress.org/support/plugin/<?php echo esc_attr($context['plugin_slug']); ?>/reviews/#new-post"
```
[the remark this answers is not in the document]
Comment (olatechpro) [reply modules/checklists/templates/footer.php]: updated

File: modules/settings/settings.php [hunk 3]
```diff
@@ -432,7 +432,7 @@ public function helper_settings_validate_and_save()
                     $_POST['_wpnonce'],
                     'edit-publishpress-settings'
                 )) {
-                wp_die(__('Cheatin&#8217; uh?'));
+                wp_die(esc_html__('Cheatin&#8217; uh?'));
```
[the remark this answers is not in the document]
Comment (olatechpro) [reply modules/settings/settings.php]: updated

File: modules/settings/templates/settings.php (hunk 1 above)
Comment (olatechpro) [reply modules/settings/templates/settings.php]: updated

**Review by andergmartins — APPROVED** [review]

**Pull request merged**